pkg/server/api/agent/v1/service_test.go

```diff
@@ -757,3 +714,526 @@ func TestGetAgent(t *testing.T) {
+				Step: &agentpb.RenewAgentRequest_Params{
+					Params: &agentpb.AgentX509SVIDParams{
+						Csr: csr,
+					},
+				},
+			},
+			paramsError: status.Error(codes.Internal, "failed to sign X509 SVID: X509 CA is not available for signing"),
+		},
+		{
+			name: "failed to update attested node",
+			dsError: []error{
+				nil,
+				errors.New("some error"),
+			},
+			expectLogs: []spiretest.LogEntry{
+				renewingMessage,
+				{
+					Level:   logrus.ErrorLevel,
+					Message: "Failed to update attested node",
+					Data: logrus.Fields{
+						telemetry.AgentID: agentID.String(),
+						logrus.ErrorKey:   "some error",
+					},
+				},
+			},
+			paramReq: &agentpb.RenewAgentRequest{
+				Step: &agentpb.RenewAgentRequest_Params{
+					Params: &agentpb.AgentX509SVIDParams{
+						Csr: csr,
+					},
+				},
+			},
+			paramsError: status.Error(codes.Internal, "failed to update attested node: some error"),
+		},
+		{
+			name: "ack invalid type",
+			expectLogs: []spiretest.LogEntry{
+				renewingMessage,
+				{
+					Level:   logrus.ErrorLevel,
+					Message: "Invalid argument: unnexpected ack type: *agent.RenewAgentRequest_Ack_",
+					Data: logrus.Fields{
+						telemetry.AgentID: agentID.String(),
+					},
+				},
+			},
+			paramReq: &agentpb.RenewAgentRequest{
+				Step: &agentpb.RenewAgentRequest_Params{
+					Params: &agentpb.AgentX509SVIDParams{
+						Csr: csr,
+					},
+				},
+			},
+			ackReq: &agentpb.RenewAgentRequest{
+				Step: &agentpb.RenewAgentRequest_Params{
+					Params: &agentpb.AgentX509SVIDParams{},
+				},
+			},
+			ackError: status.Error(codes.InvalidArgument, "unnexpected ack type: *agent.RenewAgentRequest_Ack_"),
+		},
+		{
+			name: "failed to update attested node after ack",
+			dsError: []error{
+				nil,
+				nil,
+				nil,
+				errors.New("some error"),
+			},
+			expectLogs: []spiretest.LogEntry{
+				renewingMessage,
+				{
+					Level:   logrus.ErrorLevel,
+					Message: "Failed to update attested node",
+					Data: logrus.Fields{
+						telemetry.AgentID: agentID.String(),
+						logrus.ErrorKey:   "some error",
+					},
+				},
+			},
+			paramReq: &agentpb.RenewAgentRequest{
+				Step: &agentpb.RenewAgentRequest_Params{
+					Params: &agentpb.AgentX509SVIDParams{
+						Csr: csr,
+					},
+				},
+			},
+			ackError: status.Error(codes.Internal, "failed to update attested node: some error"),
+		},
+	} {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			// Setup test
+			test := setupServiceTest(t)
+			defer test.Cleanup()
+
+			switch {
+			case tt.bannedNode:
+				_, err := test.ds.CreateAttestedNode(ctx, &datastore.CreateAttestedNodeRequest{
+					Node: &common.AttestedNode{
+						SpiffeId:            agentID.String(),
+						AttestationDataType: "tt",
+					},
+				})
+				require.NoError(t, err)
+			case !tt.noAttestedNode:
+				_, err := test.ds.CreateAttestedNode(ctx, &datastore.CreateAttestedNodeRequest{
+					Node: defaultNode,
+				})
+				require.NoError(t, err)
+			}
```

Comment:
nit: maybe we can just have an `createNode` field that we set to one of the following instead of having flags that don't make sense together?
1. `nil`
2. `bannedNode`
3. `goodNode`